xskillscore/core/comparative.py

```diff
@@ -0,0 +1,168 @@
+            forecast1 to be compared to observation
+        forecast2 : xarray.Dataset or xarray.DataArray
+            forecast2 to be compared to observation
+        observation : xarray.Dataset or xarray.DataArray
+            observation to be compared to both forecasts
+
+            if None, then assume that forecast1 and forecast2 have already been compared
+            to observation. Decide comparison based on ``metric`` or choose from:
+
+                * ``negatively_oriented_already_evaluated``: metric between forecast1
+                  (forecast2) and observations. Distances are positively oriented,
+                  therefore the smaller distance wins.
+                * ``positively_oriented_already_evaluated``: metric between forecast1
+                  (forecast2) and observations. The larger positively oriented metric
+                  wins.
+
+        dim : str
+            time dimension of dimension over which to compute the random walk.
+            This dimension is not reduced, unlike in other xskillscore functions.
+        alpha : float
+            significance level for random walk.
+        categorical : bool, optional
+            If True, the winning forecast is only rewarded a point if it exactly equals
+            the observations
+        metric : callable, optional
+            metric to compare forecast# and observation or that has been used to
+            compare forecast# and observation before using xs.sign_test.
+
+        Returns
+        -------
+        xarray.DataArray or xarray.Dataset reduced by dim containing the sign test and
+            confidence as new dimension results:
+
+                * ``sign_test``: positive (negative) number shows how many times over
+                    ``dim`` ``forecast1`` is better (worse) than ``forecast2``.
+                * ``confidence``: Positive boundary for the random walk at significance
+                    ``alpha``.
+
+
+        Examples
+        --------
+        >>> f1 = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> f2 = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> o = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> st = sign_test(f1, f2, o, dim='time')
+        >>> st.sel(results='sign_test').plot()
+        >>> st.sel(results='confidence').plot(c='gray')
+        >>> (-1*st.sel(results='confidence')).plot(c='gray')
+
+        References
+        ----------
+            * DelSole, T., & Tippett, M. K. (2016). Forecast Comparison Based on Random
+              Walks. Monthly Weather Review, 144(2), 615–626. doi: 10/f782pf
+    """
+    # TODO: account for forecast, observation ordering in climpred and observation, forecast ordering in xskillscore; do we need to change something? I think NO
+
+    # make sure metric is a callable
+    if isinstance(metric, str):
```

Comment:
I think metric should only be allowed to be a `function`. For users to specify `xs` metrics other than the default (`mae`) they would specify, e.g., `metric = xs.mse`

Reply:
I allow xs.metric name strings only